### ISSUE TITLE
LM25SC 16-mm focal length lens discontinued - proper replacement

### ISSUE THREAD

**Issue by aromanowski** [g2-bernotas/PS-Plant-Framework#16]
Hi Gytis!
I noticed that the Kowa 1”SC LM25SC 16-mm focal length has been discontinued. Would there be any issues if I purchased the suggested replacement LM25FC24M (see below)?
https://www.kowa-lenses.com/en/lm25fc24m-24mp-industrial-lens-c-mount

Thank you!
Andres

**Comment by gycka** [comment]
Hi, it is a shame that they have been discontinued. I have worked with a 25-mm lens, but it was another than the one in the link (but also Kowa, but designed for a 1" sensor). The lens will work; however, you must know that the focal length will be different, which means that you will be able to capture a smaller FOV than with the 16-mm lens would (I was able to image 4 Arabidopsis plants at a height of 40-50 cm or so). If you do choose a 25mm lens, navigate to the PSConfig.properties file and at the bottom, you will find:

``` Python

# e.g.16 or 25mm. Used to generate light compensation models
lens=25

```
Change the lens value to be whatever your lens focal length is (e.g. 16/25 mm). 

One major problem I can think of is that my leaf segmentation via Mask RCNN might not work very well (or at all!) on the data obtained using the 25 mm lens since the model was trained using the data with a 16mm lens (25 mm lens will capture more detailed data), but I do believe it should be just fine, but I cannot guarantee it. 

I hope this makes sense a bit. Are you coming from a biology or computer science background?

**Comment by aromanowski** [comment]
Hi Gytis,
Thanks for the reply. I come from a biology background, although I have some formal computer science training (from a BSc degree in Computer Science that I started while I was doing my PhD but never finished).

So the reason of why i asked about the 25 mm lens is because in the paper it says:

"PS-Plant consists of a machine vision NIR monochrome camera (Grasshopper3 GS3-U3–41C6NIR-C, FLIR Systems, Wilsonville, USA) with a 16-mm fixed focal length lens (**Kowa 1”SC LM25SC**, Kowa Company Limited, Yokohama, Japan)."

When I checked in Kowa, it mentioned that the LM25SC is a 25mm lens and suggested the replacement part should be LM25FC24M (which is also 25mm). However, they also list this one: https://www.kowa-lenses.com/en/lm16fc24m-24mp-industrial-lens-c-mount which is 16mm. Based on your reply, should I get this one?

Basically, I guess what I am asking is:
- Which camera + lens combination would you recommend I get for the project so that your software works as is? 

We can always meet and talk about collaboration if you are interested!

Right now I am starting some experiments with Alistair but want to have a functional rig on my end as well :)

All the best,
Andres

**Comment by gycka** [comment]
Hi Andres,

It is great to hear that you are working with Alister! Sorry that it took me so long to reply - probably you have already moved on with your decision, but if not, I would say go for the Grasshopper 3 + 16 mm lens as in that case you should be able to work with more plants, i.e., a larger field of view. 

Why don't you borrow one of the rigs from Alistair? Or is it a requirement for you to build your own/or his equipment is fully utilised?

**Comment by aromanowski** [comment]
Hey Gytis,
No worries at all. Thanks for the reply! I am using both of his rigs in Edi
atm via a colleague, and have to build one for Utrecht (where I am right
now) to do follow up work.

All the best,
Andres
